src/mlpack/methods/ann/layer/transposed_convolution_impl.hpp

```diff
@@ -60,19 +60,21 @@ TransposedConvolution<
     const size_t padW,
     const size_t padH,
     const size_t inputWidth,
-    const size_t inputHeight) :
+    const size_t inputHeight,
+    const size_t outputWidth,
+    const size_t outputHeight) :
     inSize(inSize),
     outSize(outSize),
     kW(kW),
     kH(kH),
     dW(dW),
     dH(dH),
-    padW(padW),
-    padH(padH),
+    padW(kW - padW - 1),
+    padH(kH - padH - 1),
     inputWidth(inputWidth),
     inputHeight(inputHeight),
-    outputWidth(0),
-    outputHeight(0)
+    outputWidth(outputWidth),
```

Comment:
We should update the serialisation as well.

Reply:
Sorry, but I didn't really get what you meant here.

Reply:
Agreed, should we also serialize the members `aH` and `aW` in `serialize()`?  (Also, should we initialize them here?)

Reply:
I think we should do both of those things. Thanks for pointing out.